app/models/proposal.rb

```diff
@@ -60,7 +60,10 @@ class Proposal < ActiveRecord::Base
   scope :for_state, ->(state) do
     where(state: state).order(:title).includes(:event, {speakers: :user}, :review_taggings)
   end
-  scope :in_track, ->(track) { where(track: track) }
+  scope :in_track, ->(track_id) do
+    track_id = nil if track_id.try(:strip).blank?
```

Comment:
I tested on my local, and I think `track_id.try(:strip) == ''` should make this work for string ids, integer ids, and track model instances.